crates/bevy_time/src/time.rs

```diff
@@ -92,57 +117,178 @@ impl Time {
     pub fn update_with_instant(&mut self, instant: Instant) {
-        if let Some(last_update) = self.last_update {
-            self.delta = instant - last_update;
-            self.delta_seconds_f64 = self.delta.as_secs_f64();
+        let raw_delta = if let Some(last_update) = self.last_update {
+            instant - last_update
+        } else {
+            instant - self.startup
+        };
+
+        let delta = if self.paused {
+            Duration::ZERO
+        } else if self.relative_speed != 1.0 {
+            // Avoid rounding errors when the relative speed is 1.
+            raw_delta.mul_f64(self.relative_speed)
+        } else {
+            raw_delta
+        };
+
+        if self.last_update.is_some() {
+            self.delta = delta;
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.raw_delta = raw_delta;
+            self.raw_delta_seconds = self.raw_delta.as_secs_f32();
+            self.raw_delta_seconds_f64 = self.raw_delta.as_secs_f64();
+        } else {
+            self.first_update = Some(instant);
         }
 
-        self.time_since_startup = instant - self.startup;
-        self.seconds_since_startup = self.time_since_startup.as_secs_f64();
+        self.elapsed_since_startup += delta;
+        self.seconds_since_startup = self.elapsed_since_startup.as_secs_f32();
+        self.seconds_since_startup_f64 = self.elapsed_since_startup.as_secs_f64();
+        self.raw_elapsed_since_startup += raw_delta;
+        self.raw_seconds_since_startup = self.raw_elapsed_since_startup.as_secs_f32();
+        self.raw_seconds_since_startup_f64 = self.raw_elapsed_since_startup.as_secs_f64();
         self.last_update = Some(instant);
     }
 
-    /// The delta between the current tick and last tick as a [`Duration`]
+    /// Returns the [`Instant`] the app was started.
+    #[inline]
+    pub fn startup(&self) -> Instant {
+        self.startup
+    }
+
+    /// Returns the [`Instant`] when [`update`](Self::update) was first called, if it exists.
+    #[inline]
+    pub fn first_update(&self) -> Option<Instant> {
+        self.first_update
+    }
+
+    /// Returns the [`Instant`] when [`update`](Self::update) was last called, if it exists.
+    #[inline]
+    pub fn last_update(&self) -> Option<Instant> {
+        self.last_update
+    }
+
+    /// Returns the rate that time advances relative to raw CPU time, as [`f32`].
+    #[inline]
+    pub fn relative_speed(&self) -> f32 {
+        self.relative_speed as f32
+    }
+
+    /// Returns the rate that time advances relative to raw CPU time, as [`f64`].
+    #[inline]
+    pub fn relative_speed_f64(&self) -> f64 {
+        self.relative_speed
+    }
+
+    /// Sets the rate that time advances relative to raw CPU time, given as [`f32`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if `ratio` is negative or not finite.
+    pub fn set_relative_speed(&mut self, ratio: f32) {
```

Comment:
How about `set_time_scale`?
Unity and Godot call it time scale and Unreal calls it time dilation.

Reply:
I only referenced other engines to show the universal presence of the feature. IMO they've all named it incorrectly.

Increasing time dilation or scale would mean to lengthen each unit of time (i.e. slow things down), which is the inverse of how we actually use the values (i.e. if dilation is 1.1x, time should pass at 1.0/1.1 speed, but this PR and other engines actually go faster).

I think "relative speed" is the most literal term here.

Reply:
Huh, I suppose you're right. Though I'm not sure being correct is more valuable than the option being recognizable.

Reply:
Might be worth adding a line saying something like "Sometimes referred to as time scaling or time dilation in other engines"